runtime/gc_base/ScavengerForwardedHeader.hpp

```diff
@@ -46,13 +49,29 @@
  */
 class MM_ScavengerForwardedHeader
 {
-public:
+private:
 protected:
 	omrobjectptr_t _objectPtr; /**< the object on which to act */
 	uintptr_t _preserved; /**< a backup copy of the header fields which may be modified by this class */
 #if defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS)
 	bool const _compressObjectReferences;
 #endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
-private:
+public:
+	/**
+	 * Construct a new ScavengerForwardedHeader for the specified object.
+	 * We need a constructor for backward compatibility.
+	 *
+	 * @parm[in] object the object to be used
+	 * @parm[in] extensions the current GC extensions
+	 */
+	MMINLINE
+	MM_ScavengerForwardedHeader(omrobjectptr_t object, MM_GCExtensionsBase *extensions) :
+		_objectPtr(object)
+		, _preserved(*(volatile uintptr_t *)_objectPtr)
+#if defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS)
+		, _compressObjectReferences(extensions->compressObjectReferences())
+#endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
+	{
+	}
```

Comment:
With changes to the other files, this constructor doesn't appear to be used. Why add it then?

Reply:
`JDK11_x86-64_mac_mixed` seems to be complaining about it:
```
22:21:57  In file included from /Users/jenkins/workspace/Build_JDK11_x86-64_mac_mixed_Nightly/openj9/runtime/ddr/gcddr.cpp:38:
22:21:57  /Users/jenkins/workspace/Build_JDK11_x86-64_mac_mixed_Nightly/openj9/runtime/gc_base/ScavengerForwardedHeader.hpp:47:7: error: class 'MM_ScavengerForwardedHeader' does not declare any constructor to initialize its non-modifiable members [-Werror]
22:21:57  class MM_ScavengerForwardedHeader
22:21:57        ^
22:21:57  /Users/jenkins/workspace/Build_JDK11_x86-64_mac_mixed_Nightly/openj9/runtime/gc_base/ScavengerForwardedHeader.hpp:54:13: note: const member '_compressObjectReferences' will never be initialized
22:21:57          bool const _compressObjectReferences;
22:21:57                     ^
22:21:57  1 error generated.
```
Here: https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_mac_mixed_Nightly/7/

Reply:
Right, this class only exists for backwards compatibility of DDR. Thanks.

Reply:
Correct, sorry forgot to mention that

Reply:
Actually, I think I'd prefer that we don't add the constructor. Instead, simply remove `const` from the declaration of `_compressObjectReferences`.